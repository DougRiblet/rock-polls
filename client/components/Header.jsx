// @flow

import React from 'react';
import { Link } from 'react-router-dom';

const Header = () => (
  <header>
    <nav>
      <div id='header-full'>
        <div id='header-container'>
          <div className='nav-left'>
            <ul>
              <li><Link to='/'>Home</Link></li>
              <li><Link to='/polls'>Polls</Link></li>
            </ul>
          </div>
          <div className='nav-right'>
            <ul>
              <li><Link to='/admin'>Admin</Link></li>
              <li><Link to='/login'>Login</Link></li>
              <li><Link to='/logout'>Logout</Link></li>
              <li><Link to='/signup'>Signup</Link></li>
            </ul>
          </div>
        </div>
      </div>
    </nav>
  </header>
);

export default Header;
