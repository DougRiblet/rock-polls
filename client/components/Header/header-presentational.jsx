// @flow

import React from 'react';
import { Link } from 'react-router-dom';

type Props = {
  authenticated: boolean, 
};

const NavLoggedIn = () => (
  <ul>
    <li><Link to='/admin'>Admin</Link></li>
    <li><Link to='/admin/create'>Create Poll</Link></li>
    <li><Link to='/logout'>Logout</Link></li>
  </ul>
);

const NavLoggedOut = () => (
  <ul>
    <li><Link to='/login'>Login</Link></li>
    <li><Link to='/signup'>Signup</Link></li>
  </ul>
);

const Header = ({ authenticated }: Props) => (
  <header>
    <nav>
      <div id='header-full'>
        <div id='header-container'>
          <div id='nav-left'>
            <ul>
              <li><Link to='/'>Home</Link></li>
              <li><Link to='/polls'>Polls</Link></li>
            </ul>
          </div>
          <div id='nav-right'>
            {
              authenticated
              ? <NavLoggedIn />
              : <NavLoggedOut />
            }
          </div>
        </div>
      </div>
    </nav>
  </header>
);

export default Header;
