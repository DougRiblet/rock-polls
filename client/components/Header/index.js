import { connect } from 'react-redux';
import headerPresent from './header-presentational';

const mapStateToProps = state => ({
  authenticated: state.auth.authenticated,
});

export default connect(mapStateToProps)(headerPresent);
