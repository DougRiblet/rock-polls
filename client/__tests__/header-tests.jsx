import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Header from '../components/Header/header-presentational';

jest.mock('react-router-dom', () => ({ Link: 'Link' }))

describe('Header', () => {
  test('Snapshot', () => {
    const tree = renderer.create(<Header
      authenticated={false}
    />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Header
      authenticated={false}
    />, div);
  });
});
