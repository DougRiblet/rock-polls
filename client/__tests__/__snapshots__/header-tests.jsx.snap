// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Snapshot 1`] = `
<header>
  <nav>
    <div
      id="header-full"
    >
      <div
        id="header-container"
      >
        <div
          id="nav-left"
        >
          <Link
            to="/"
          >
            Coding Polls
          </Link>
        </div>
        <div
          id="nav-right"
        >
          <div>
            <Link
              to="/login"
            >
              login
            </Link>
            <Link
              to="/signup"
            >
              signup
            </Link>
          </div>
        </div>
      </div>
    </div>
  </nav>
</header>
`;
